django/db/models/query.py

```diff
@@ -454,6 +454,8 @@ def bulk_create(self, objs, batch_size=None):
                         assert len(ids) == len(objs_without_pk)
                     for i in range(len(ids)):
                         objs_without_pk[i].pk = ids[i]
+                        objs_without_pk[i]._state.adding = False
+                        objs_without_pk[i]._state.db = self.db
```

Comment:
It might be worth storing `objs_without_pk[i]` in an intermediary variable to avoid 3 key access.

Using `zip` would look be even better IMHO:

```python
db = self.db
for obj_without_pk, pk in zip(objs_without_pk, ids):
    obj_without_pk.pk = pk
    obj_without_pk._state.adding = False
    obj_without_pk._state.db = db
```